docker/Dockerfile

```diff
@@ -0,0 +1,69 @@
+FROM debian:11-slim as build
+
+WORKDIR /build
+
+RUN --mount=type=cache,target=/var/cache/apt \
+    echo "Install build dependencies" && \
+    apt update && \
+    apt install gcc cmake zlib1g-dev make build-essential g++ unzip ca-certificates wget -yqq --no-install-recommends && \
+    rm -rf /var/lib/apt/lists/* && \
+    update-ca-certificates && \
+    wget https://dev.mysql.com/get/Downloads/Connector-C++/libmysqlcppconn9_8.0.27-1debian11_amd64.deb -O /tmp/libmysqlcppconn.deb && \
+    dpkg -i /tmp/libmysqlcppconn.deb && \
+    rm /tmp/libmysqlcppconn.deb
+
+COPY dAuthServer/ /build/dAuthServer
+COPY dChatServer/ /build/dChatServer
+COPY dCommon/ /build/dCommon
+COPY dChatFilter/ /build/dChatFilter
+COPY dDatabase/ /build/dDatabase
+COPY dGame/ /build/dGame
+COPY dMasterServer/ /build/dMasterServer
+COPY dNet/ /build/dNet
+COPY dPhysics/ /build/dPhysics
+COPY dScripts/ /build/dScripts
+COPY dWorldServer/ /build/dWorldServer
+COPY dZoneManager/ /build/dZoneManager
+COPY migrations/ /build/migrations
+COPY resources/ /build/resources
+COPY thirdparty/ /build/thirdparty
+COPY vanity /build/vanity
+COPY .clang-* CMake* LICENSE /build/
+
+ARG BUILD_THREADS=1
+ARG BUILD_VERSION=171022
+
+RUN echo "Build server" && \
+    mkdir -p build && \
+    cd build && \
+    sed -i -e "s/171023/${BUILD_VERSION}/g" ../CMakeVariables.txt && \ 
+    cmake .. && \
+    make -j $BUILD_THREADS
+
+RUN mkdir -p /build/build/res/maps/navmeshes/ && \
+    unzip /build/resources/navmeshes.zip -d /build/build/res/maps
+
+FROM debian:11-slim as runtime
```

Comment:
I'd actually recommend one more layer. This layer is responsible for extracting and setting up the runtime data, but it has a bunch of dependencies not needed for runtime (ie. python3 sqlite3). Maybe use a python layer here (skips needing to install python3), run the migrations/extraction, then do your runtime layer without any of those other dependencies as a debian slim.

Reply:
Seconded. I think once the build is finished, putting everything in an alpine contain (for example) necessary to run is the way to go. Not sure if alpine needs libstdc++ updated / installed though... that will be something to check on. And ofc any dynamic libraries should be installed if necessary.

Reply:
> Seconded. I think once the build is finished, putting everything in an alpine contain (for example) that's necessary to run is the way to go. Not sure if alpine needs libstdc++ updated / installed though... that will be something to check on. And ofc any dynamic libraries should be installed if necessary.

A few moments ago I actually tried to make an alpine image from this. There were some problems. I also would like to have an alpine image instead of debian.

Reply:
Yeah I tried alpine and also had issues. I wonder if the binary has some dependencies that alpine doesn't support out of the box.

Reply:
I tried compiling it using alpine and ran into a bunch of errors with [these](https://github.com/DarkflameUniverse/DarkflameServer/blob/0db954fa72a76b2640dfebcd7987ac3e2ebe0a46/dDatabase/Tables/CDTable.h#L20) [lines](https://github.com/DarkflameUniverse/DarkflameServer/blob/0db954fa72a76b2640dfebcd7987ac3e2ebe0a46/dCommon/dCommonVars.h#L55) because alpine uses musl. Alpine also doesn't have backtrace used [here](https://github.com/DarkflameUniverse/DarkflameServer/blob/3c3a3dba8fa6fc1d3a034e34a96c3f4fff88d93f/dCommon/Diagnostics.cpp#L61).

Reply:
Hmmm... does `ldd ./MasterServer` print anything out? I'm curious to see what it's linking against (at work atm so I don't have access myself)

Reply:
Maybe we can try this in a followup PR.